recipes-kernel/kernel-tests/kernel-performance-tests-files/run-cyclictest

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+source ./common.cfg
+
+function add_system_info()
+{
+    {
+	DESC=$(fw_printenv DeviceDesc)
+	DEVICE=${DESC#*=}
+	echo "# Device: $DEVICE"
+
+	echo -n "# Kernel: "
+	uname -a
+
+	echo -n "# Kernel parameters: "
+	cat /proc/cmdline
+
+	# save info on the current security mitigations
+	echo "# Security vulnerabilities settings: "
+	if [ -d "/sys/devices/system/cpu/vulnerabilities" ]; then
+	    for f in  /sys/devices/system/cpu/vulnerabilities/*; do
+		VULN=$(basename "$f")
+		if [ -f "$f" ]; then
+		    echo -n -e "#     $VULN:\t"
+		    cat "$f"
+		fi
+	    done
+	else
+	    echo "#     vulnerability information not published"
+	fi
+
+	# save info on the current C state settings
+	echo "# C-states settings: "
+	for CPU in /sys/devices/system/cpu/cpu[0-9]; do
+	    NCPU=$(basename "$CPU")
+	    echo "#     [$NCPU]"
+	    for CSTATE in $CPU/cpuidle/state[0-9]; do
+		NSTATE=$(basename "$CSTATE")
+		if [ -f "$CSTATE/name" ] && [ -f "$CSTATE/disable" ]; then
+                    NAME=$(cat "$CSTATE/name" 2>/dev/null)
+                    STATUS=$(cat "$CSTATE/disable" 2>/dev/null)
+
+                    echo -n "#     $NSTATE: $NAME: "
+                    if [ "$STATUS" = "0" ]; then
+                        echo "enabled"
+                    else
+                        echo "disabled"
+                    fi
+		else
+                    echo "#     information not available"
+		fi
+	    done
+	done
+    } >> "$1"
+}
+
+function run_cyclictest()
+{
+    LOG="results/cyclictest-$1-`date +'%Y_%m_%d-%H_%M_%S'`.log"
+    cyclictest --smp --priority=98 --mlockall --notrace --interval=997 --quiet --duration="$TEST_DURATION" --histofall=1000 --histfile="$LOG" > /dev/null
+    add_system_info "$LOG"
+
+    LATENCY=$(grep -sw "# Max Latencies:" "$LOG" | awk '{max=$4; for(i=4; i<=NF; i++) if ($i>max) max=$i; gsub("^0*", "", max); print max}')
+    if [ "$LATENCY" -le "$MAX_LATENCY" ]; then
+	echo "PASS: cyclictest with $1 load, latency: $LATENCY (usec) is less than max latency: $MAX_LATENCY (usec)"
+	CYCLICTEST_RESULT=0
+    else
+	echo "FAIL: cyclictest with $1 load, latency: $LATENCY (usec) is above the expected max latency: $MAX_LATENCY (usec)"
```

Comment:
For ptest parsing, the contents of the results line needs to be consistent between runs (with the exception of the actual PASS/FAIL substring). I cover that in more detail [here](https://dev.azure.com/ni/DevCentral/_wiki/wikis/AppCentral.wiki/3560/Yocto-PTests?anchor=test-result-lines). You can print the load and latency info on lines before the PASS/FAIL line, though.

Reply:
Sure; I didn't realize there was convention doc for this. That's good to know. The Yocto manual doesn't mention anything beyond the PASS/FAIL substring.

Reply:
Yep. Believe me, that wiki page is written in blood. :)